### PR TITLE
Use moto for dynamodb test cases, fixes #149 and #150

### LINKED ISSUE #149: Add database mocking when testing
**Describe what needs to be changed.**
Currently test_database.py creates records on the production database.

**Describe the proposed changes**
When testing, mock a database and check the response.

**Additional context**
https://github.com/spulec/moto

_Originally posted by @Exsects in https://github.com/mbox-contributors/mbox/issues/142#issuecomment-877543085_

### LINKED ISSUE #150: Separate unit test functionality in test_database.py
**Describe what needs to be changed.**
Make unit tests test only one part at a time.

**Describe the proposed changes**
Change test record function to only test one function of the record class at a time.

_Originally posted by @Exsects in https://github.com/mbox-contributors/mbox/issues/142#issuecomment-877543085_

### PR DESCRIPTION
**Pull request by Exsects**
---
name: Feature request
about: Suggest an idea for this project
title: ''
labels: enhancement
assignees: ''

---

**Description**

> [NOTE]
> Please provide list of issues that are closed by this PR.
<https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue>

A few sentences describing the overall goals of the pull request's commits.
- Adds back test_database.py which was previously deleted. 
- Fixes #149 and #150 

**Changes include**

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Refactor (non-breaking/breaking change that refactor the previous code base)
- [x] Tests

**Impacted Areas in Application**

List general components of the application that this PR will affect:
None

**Todos**

- [ ] Updated and Passed Tests
- [ ] Passed Quality Checks
- [ ] Updated  Documentation

**Screenshots**

Add any screenshots as needed

**Other comments**

### CODE REVIEW

**Review by AvJain — APPROVED** [review]
Changes look fine to me

File: tests/unit/test_database.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import os
+from typing import List
+
+import boto3
+import pytest
+from moto import mock_dynamodb2
+
+from src.element.database import DynamoDB, Record
+
+sample_record_list: List[Record] = [
```
Comment (AvJain) [inline tests/unit/test_database.py]: This is fine for now, but ideally we should think about creating a method that will make the creation of this object easier in the future. Something like create_dynamo_record_for_testing( ... ) If we plan on adding more tests to this file in the future adding the method mentioned above will make things easier to follow and read

**Review by AvJain — DISMISSED** [review]

**Review by Karan-Jain — COMMENTED** [review]
Move the fixture functions to the conftest file. Imports will require no import  statements. You will just need to pass them as arguments like you are doing now

File: tests/unit/test_database.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            ProvisionedThroughput={
+                "ReadCapacityUnits": 10,
+                "WriteCapacityUnits": 10,
+            },
+        )
+
+        table = boto3.resource("dynamodb", region_name="us-east-2").Table(
+            "music-box-db"
+        )
+
```
Comment (Karan-Jain) [inline tests/unit/test_database.py]: Do you always want two items in the table? You can potenially move this to a fixture.

File: tests/unit/test_database.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+@pytest.fixture(scope="function")
+def aws_credentials():
+    """Mocked AWS Credentials for moto."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
+    os.environ["AWS_SECURITY_TOKEN"] = "testing"
+    os.environ["AWS_SESSION_TOKEN"] = "testing"
+
+
+@pytest.fixture(scope="function")
```
Comment (Karan-Jain) [inline tests/unit/test_database.py]: Move all fixtures to conftest

File: tests/unit/test_database.py (hunk 1 above)
Comment (Exsects) [reply tests/unit/test_database.py]: Yes I agree with your suggestion. By any chance, do you know where/how we should put all the sample data?

**Comment by Exsects** [comment]
> Move the fixture functions to the conftest file. Imports will require no import  statements. You will just need to pass them as arguments like you are doing now
> 

Will do more research and take a look it. Thanks!

**Review by borisliao — APPROVED** [review]

**Review by Karan-Jain — APPROVED** [review]

**Review by AvJain — APPROVED** [review]

**Pull request merged**